include/fiction/algorithms/simulation/sidb/maximal_defect_influence_distance.hpp

```diff
@@ -0,0 +1,260 @@
+//
+// Created by Jan Drewniok on 21.06.23.
+//
+
+#ifndef FICTION_MAXIMAL_DEFECT_INFLUENCE_DISTANCE_HPP
+#define FICTION_MAXIMAL_DEFECT_INFLUENCE_DISTANCE_HPP
+
+#include "fiction/algorithms/simulation/sidb/critical_temperature.hpp"
+#include "fiction/algorithms/simulation/sidb/quickexact.hpp"
+#include "fiction/layouts/bounding_box.hpp"
+#include "fiction/technology/sidb_defects.hpp"
+#include "fiction/technology/sidb_surface.hpp"
+#include "fiction/utils/layout_utils.hpp"
+
+#include <mutex>
+#include <thread>
+#include <utility>
+
+namespace fiction
+{
+/**
+ * This struct stores the parameters for the `maximal_defect_influence_distance` algorithm.
+ */
+template <typename Lyt>
+struct maximal_defect_influence_distance_params
+{
+    /**
+     * The defect used to calculate the maximal defect influence distance.
+     */
+    sidb_defect defect{};
+    /**
+     * Physical simulation parameters.
+     */
+    sidb_simulation_parameters physical_params{};
+    /**
+     * The coordinate describes the width and height of the area around the gate, which is
+     * also used to place defects (given in siqad coordinates).
+     * */
+    coordinate<Lyt> additional_scanning_area{50, 6};
+    /**
+     * Number of threads to spawn. By default the number of threads is set to the number of available hardware threads.
+     */
+    uint64_t number_threads{std::thread::hardware_concurrency()};
+};
+
+/**
+ * This function determines the maximum distance at which a placed defect can still affect the layout (i.e. different
+ * ground state). This means that a defect must be placed further away from the layout than the given distance in order
+ * not to change the layout's ground state.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout for which the influence distance is simulated.
+ * @param sim_params Parameters which are used to determine the defect influence distance.
+ * @return Pair of the maximum influence distance (i.e. the defect should be placed further away than this value to
+ * ensure correct behavior of e.g. a gate) with the corresponding position of the defect (helps to identify the
+ * sensitive part of the layout).
+ */
+template <typename Lyt>
+std::pair<double, typename Lyt::cell>
+maximal_defect_influence_distance(Lyt& lyt, const maximal_defect_influence_distance_params<Lyt>& sim_params)
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+    static_assert(has_siqad_coord_v<Lyt>, "Lyt is not based on SiQAD coordinates");
+
+    const quickexact_params<sidb_defect_layout> params_defect{sim_params.physical_params,
+                                                              automatic_base_number_detection::OFF};
+
+    coordinate<Lyt> min_defect_position{};
+    double          avoidance_distance = 0;
```

Comment:
If I'm not mistaken, these are the only two shared resources. You could wrap them in `std::atomic`s to prevent yourself from having to worry about locking at all.

Reply:
good idea. Thanks.